core/src/main/scala/cell/HandlerPool.scala

```diff
@@ -134,7 +133,10 @@ class HandlerPool(parallelism: Int = 8, unhandledExceptionHandler: Throwable =>
     val key = cells.head.key
     val result = key.resolve(cells)
 
-    for ((c, v) <- result) c.resolveWithValue(v)
+    for ((c, v) <- result) {
+      cells.foreach(c.removeNextCallbacks(_))
+      c.resolveWithValue(v)
+    }
```

Comment:
I think we should first remove all next callbacks and then, in a second iteration, resolve all cells in `result`. Otherwise, I think there could be a situation where `resolveWithValue` would still lead to the unwanted invocation of a next callback. Could you double-check this?

Reply:
Before https://github.com/JanKoelzer/reactive-async/commit/ecd475998bfea03a0284f136239d0da0b4355a04, I had the callbacks removed beforehand. I moved this into the loop, because it seemed more efficient. I am still positiv this works well:
c.resolveWith(v) would trigger all nextCallbacks registered in c. But those have been removed just before.
I added two tests here: https://github.com/JanKoelzer/reactive-async/blob/jan-removeCycleDeps/core/src/test/scala/cell/base.scala
They are called "whenNext: cSCC with constant resolution" and "whenNext: cSCC with default resolution".

But maybe I am missing a case. We will definitely be safe with the order you suggest. So if you do not like this version, I am going to change it. The gain is minor, anyway.

(The rest of the requested changes is already included.)

Reply:
OK, I agree now that it's safe to remove the callbacks and resolve the cell within the same loop.